### PR TITLE
Add endpoint to download the agent binaries

### LINKED ISSUE #2523: Add endpoint to download the agent binaries

### PR DESCRIPTION
**Pull request by chkeita**
## Summary of the Pull Request

closes #2523

### CODE REVIEW

File: src/ApiService/IntegrationTests/ToolsTests.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    [Fact]
+    public async Async.Task CanDownload() {
+
+        const int NUMBER_OF_FILES = 20;
+
+        var toolsContainerClient = GetContainerClient(WellKnownContainers.Tools);
+        _ = await toolsContainerClient.CreateIfNotExistsAsync();
+
+        // generate random content 
+        var files = Enumerable.Range(0, NUMBER_OF_FILES).Select((x, i) => (path: i, content: Guid.NewGuid())).ToList();
```
Comment (stishkin) [inline src/ApiService/IntegrationTests/ToolsTests.cs:43]: use more realistic content for testing ?

File: src/ApiService/ApiService/Functions/Tool.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    private static string ReadResource(Assembly asm, string resourceName) {
+        using var r = asm.GetManifestResourceStream(resourceName);
+        if (r is null) {
+            return "unknown";
+        }
+
+        using var sr = new StreamReader(r);
+        return sr.ReadToEnd().Trim();
```
Comment (stishkin) [inline src/ApiService/ApiService/Functions/Tool.cs:25]: read resource here means read manifest ?

Comment (stishkin) [reply src/ApiService/ApiService/Functions/Tool.cs:25]: does not seem to be called by anything

File: src/ApiService/IntegrationTests/ToolsTests.cs (hunk 1 above)
Comment (chkeita) [reply src/ApiService/IntegrationTests/ToolsTests.cs:43]: why ? isn't this enough to test the functionality ?

File: src/ApiService/ApiService/Functions/Tool.cs (hunk 2 above)
Comment (chkeita) [reply src/ApiService/ApiService/Functions/Tool.cs]: forgot the remove after copying the code

**Review by stishkin — APPROVED** [review]

**Pull request merged**